site/app/controllers/ChatroomController.php

```diff
@@ -0,0 +1,187 @@
+
+        $em->persist($chatroom);
+        $em->flush();
+
+        $this->core->addSuccessMessage("Chatroom successfully added");
+        return new RedirectResponse($this->core->buildCourseUrl(['chat']));
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/chat/deleteChatroom", methods={"POST"})
+     * @AccessControl(role="INSTRUCTOR")
+     */
+    public function deleteChatroom(): JsonResponse {
+        $chatroom_id = intval($_POST['chatroom_id'] ?? -1);
+        $em = $this->core->getCourseEntityManager();
+
+        /** @var \app\repositories\chat\ChatroomRepository $repo */
+        $repo = $em->getRepository(Chatroom::class);
+
+        /** @var Chatroom|null */
+        $chatroom = $repo->findByChatroomId($chatroom_id);
+
+        if ($chatroom === null) {
+            return JsonResponse::getFailResponse('Invalid Chatroom ID');
+        }
+
+        foreach ($chatroom->getMessages() as $message) {
+            $em->remove($message);
+        }
+        $em->remove($chatroom);
+        $em->flush();
+
+        return JsonResponse::getSuccessResponse();
+    }
+
+/**
+ * @Route("/courses/{_semester}/{_course}/chat/{chatroom_id}/edit", name="edit_chatroom", methods={"POST"})
+ * @AccessControl(role="INSTRUCTOR")
+ */
+public function editChatroom(int $chatroom_id): RedirectResponse {
+    $em = $this->core->getCourseEntityManager();
+    $chatroom = $em->getRepository(Chatroom::class)->find($chatroom_id);
+
+    if (!$chatroom) {
+        $this->core->addErrorMessage("Chatroom not found");
+        return new RedirectResponse($this->core->buildCourseUrl(['chat']));
+    }
+
+    if (isset($_POST['title'])) {
+        $chatroom->setTitle($_POST['title']);
+    }
+    if (isset($_POST['description'])) {
+        $chatroom->setDescription($_POST['description']);
+    }
+
+    $em->persist($chatroom);
+    $em->flush();
+
+    $this->core->addSuccessMessage("Chatroom successfully updated");
+    return new RedirectResponse($this->core->buildCourseUrl(['chat']));
+}
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/chat/{chatroom_id}/messages", name="fetch_chatroom_messages", methods={"GET"})
+     */
+    public function fetchMessages(int $chatroom_id): JsonResponse {
+        $em = $this->core->getCourseEntityManager();
+        $messages = $em->getRepository(Message::class)->findBy(['chatroom' => $chatroom_id], ['timestamp' => 'ASC']);
+
+        $formattedMessages = array_map(function ($message) {
+            return [
+                'id' => $message->getId(),
+                'content' => $message->getContent(),
+                'timestamp' => $message->getTimestamp()->format('Y-m-d H:i:s'),
+                'user_id' => $message->getUserId(),
+                'display_name' => $message->getDisplayName(),
+                'role' => $message->getRole()
+            ];
+        }, $messages);
+
+        return JsonResponse::getSuccessResponse($formattedMessages);
+    }
```

Comment:
Will there be a restriction for only messages from that day?

Reply:
Good question. Currently there's no restriction to a student's access to the history of messages. We could either set this restriction to either day-based, session-based, or decide something else.